parquet-column/src/main/java/org/apache/parquet/column/statistics/BinaryStatistics.java

```diff
@@ -46,8 +46,8 @@ public void mergeStatisticsMinMax(Statistics stats) {
 
   @Override
   public void setMinMaxFromBytes(byte[] minBytes, byte[] maxBytes) {
-    max = Binary.fromByteArray(maxBytes);
-    min = Binary.fromByteArray(minBytes);
+    max = Binary.fromReusedByteArray(maxBytes);
+    min = Binary.fromReusedByteArray(minBytes);
```

Comment:
I think it could be debated from either ways. A producer might want to just update the max/ min bytes, instead of explicitly calling updateStats by creating a new Binary every time it has to update. In this case, if we add a comment stating the behavior explicitly will be nice. I am doing that for now. If you disagree, then I will change it copy instead.